Sources/AWSSDKSwift/Middlewares/S3/S3RequestMiddleware.swift

```diff
@@ -15,6 +15,7 @@ public struct S3RequestMiddleware: AWSRequestMiddleware {
 
         switch request.httpMethod.lowercased() {
         case "get":
+            guard let host = request.url.host, host.contains("amazonaws.com") else { break }
```

Comment:
🤔 on line 21 we check for this, and support when it's not true. Is there a valid use case where the host does not contain `amazonaws`?

Reply:
right. So then what is the purpose of the code on lines 21-26 after this change?

Reply:
I guess they can be removed, I'll do that

Reply:
Removed this change so we can submit this PR. Will add another PR for this